### PR TITLE
create-astro updates

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

This PR includes a few changes to the experience of creating a new project in Astro:

- [x] The list of available templates is now the following:
  - Basics
  - Blog
  - Docs (Starlight)
  - Portfolio
  - In the stable version of Astro 5, Storefront will be added as well
- [x] The TypeScript question has been removed, the default value is now strict for everyone. The presets still exists for anyone wanting a more relaxed or strict experience. Remember, Astro is TypeScript.
  - [x] `// @ts-check` has been added to all the examples config files
  - [x] `astro check` is no longer enabled by default for the build command
- [x] `create-astro` now takes a `--add` flag, allowing one to add a bunch of integrations in one command.

### Basic example rework

The basics template has been reworked to provide a more minimal template, allowing it to serve as a mix between "Basics" and "Minimal":

- The theme of the basics template has been moved into a built-in "Welcome" component exported from Astro directly. The source code of that component is plain `.astro`, allowing one to still inspect how you'd achieve things in Astro. This component takes in a props to customise the title, still allowing for the implicit teaching of props. A very basic layout (similar to Emmet's `doc` command) is included, both to setup basic HTML stuff and teach the user about templates / slots

## Testing

Added / updated tests

## Docs

We'll update the docs:tm:

### CODE REVIEW

**Comment by TheOtterlord** [comment]
Todo:
- [ ] Add tests for `--add`
- [ ] Add documentation for `--add`
- [ ] No longer set to `output: 'server'` when using `astro add`?

File: packages/astro/components/index.ts
Comment (florian-lefebvre) [inline packages/astro/components/index.ts:8]: Maybe Welcome instead of WelcomeScreen?

File: .changeset/old-actors-learn.md
Comment (florian-lefebvre) [inline .changeset/old-actors-learn.md:5]: Need to talk about TS changes as well

File: packages/create-astro/src/actions/template.ts
Comment (TheOtterlord) [inline packages/create-astro/src/actions/template.ts:26]:
```suggestion
				{ value: 'starlight', label: 'Use docs (Starlight) template' },
```

File: packages/astro/components/WelcomeScreen.astro
Comment (bholmesdev) [inline packages/astro/components/WelcomeScreen.astro:1]: I think moving the original design here was smart for keeping this PR reviewable! Still, I recall that @FredKSchott wanted to use this as an opportunity to explore a new, more expressive design.

May become a blocker to make the new starter more of a _feature_ we promote. Still on the fence whether merging this change in code structure without a redesign is a good idea.

File: .changeset/old-actors-learn.md
Comment (bholmesdev) [reply .changeset/old-actors-learn.md:5]: I think `--add` should be split to a separate PR to avoid blocking on this review. Sounds it like we can release that independently.

File: packages/astro/components/WelcomeScreen.astro
Comment (Princesseuh) [reply packages/astro/components/WelcomeScreen.astro:1]: For what it's worth, this is on the next branch, so it would only affect new projects using the beta. Even if we merge this right now, there's plenty of time to make the new design before the actual release.

File: .changeset/old-actors-learn.md
Comment (sarah11918) [inline .changeset/old-actors-learn.md:5]:
```suggestion
Reworks the experience of creating a new Astro project using the `create astro` CLI command.
```

File: .changeset/old-actors-learn.md
Comment (sarah11918) [inline .changeset/old-actors-learn.md:7]:
```suggestion
- Updates the list of templates to include Starlight and combines the "minimal" and "basics" templates into a new, refreshed "Basics" template to serve as the single, minimal Astro project starter.
```
Maybe something like this is fine here, and then we don't need the paragraph at the end?

File: .changeset/old-actors-learn.md
Comment (sarah11918) [inline .changeset/old-actors-learn.md:8]:
```suggestion
- Removes the TypeScript question. Astro is TypeScript-only, so this question was often misleading. The "Strict" preset is now the default, but it can still be changed manually in `tsconfig.json`. 
- `astro check` is no longer automatically added to the build script.
```
I know they're both TypeScript related, but it seemed like there were different, separate reasons for removing `astro check` so this probably deserves its own line? And, we could put an explanation of why if you wanted!

File: .changeset/old-actors-learn.md
Comment (sarah11918) [inline .changeset/old-actors-learn.md:9]:
```suggestion
- Adds a new question allowing the user to choose an adapter for their project, if they already know where they want to deploy it.
```

File: .changeset/old-actors-learn.md
Comment (sarah11918) [inline .changeset/old-actors-learn.md:11]:
```suggestion
```
We can probably get away without needing another statement here if we choose to work something into the list item above!

File: packages/create-astro/src/actions/dependencies.ts
Comment (sarah11918) [inline packages/create-astro/src/actions/dependencies.ts:56]:
```suggestion
					{ value: undefined, label: 'none' },
```
*Maybe* lower-case "none" would be easier to read and distinguish from "Node", which look quite close. 😅  Also, "none" is not a proper name, so no need to capitalize here, and I think that's a good thing!

File: packages/create-astro/src/actions/dependencies.ts
Comment (sarah11918) [inline packages/create-astro/src/actions/dependencies.ts:54]:
```suggestion
				message: 'Add an official adapter',
```
Don't really feel strongly about this, but more noting that since there *are* community adapters, we *might* get PRs or issues about "can I get my adapter added here?" that we can avoid by either something like adding "official" as shown above 

OR, after "none", a message pointing to the integrations catalogue for community adapters like,. "No adapter added. You can also find community adapters to install at https://astro.build/integrations!" or something like that. (The way we used to say, "No problem! about Typescript)

**Review by sarah11918 — COMMENTED** [review]
Thanks for pinging, Erika!

So, off the top of my head (I'll make a task for myself to follow up):

- We don't *need* anything in the Upgrade guide since technically this can't break an existing project. I suspect this will either be more of a "discovery" or something to hype for the blog post.
- I will need to update the tutorial that is currently expecting the `base` TS setting, and the current "minimal" template 
- I will check the TypeScript/config pages that describe our settings to make sure nothing *expects* base and describes updating that
- document the new `--add` flag

- As much as we can manage (which might not be much depending on our whole sidebar recreation? This might have to be the next quarter project), we can look at updating code examples with Tabs for JS/TS, or showing more TS examples in docs now that we can expect the basic project to use `strict` and more people to need to avoid squiggles. 😅  I will see if I can wrangle some maintainers/contributors to start prepping for this at least.

File: packages/create-astro/src/actions/template.ts
Comment (sarah11918) [inline packages/create-astro/src/actions/template.ts:24]:
```suggestion
				{ value: 'basics', label: 'A basic, minimal starter', hint: '(recommended)' },
```
Again, don't feel strongly, but I feel like "include sample files" was probably in part to distinguish it from "empty" project, and now this might feel like "But, I don't want your files, I want a new fresh project from scratch!"

So, maybe highlighting somehow that this *is* honestly a minimal starter could be helpful?

**Review by ematipico — APPROVED** [review]

File: .changeset/old-actors-learn.md
Comment (Princesseuh) [reply .changeset/old-actors-learn.md:11]: I agree!

File: packages/create-astro/src/actions/dependencies.ts
Comment (Princesseuh) [reply packages/create-astro/src/actions/dependencies.ts:54]: That's fine with me, since there's not many community adapter I think we can consider adding the message separately from this PR

File: packages/create-astro/src/actions/dependencies.ts
Comment (Princesseuh) [reply packages/create-astro/src/actions/dependencies.ts:56]: Hmm, I admit that it looks weird to have None and Node next to each others, but it also looks weird in lower case, since every other options (both in this prompt and other) always start with an uppercase 🤔

**Review by florian-lefebvre — APPROVED** [review]

File: packages/create-astro/src/actions/dependencies.ts
Comment (bluwy) [reply packages/create-astro/src/actions/dependencies.ts:56]: Perhaps we can rename `Node` to `Node.js`?

File: packages/create-astro/src/actions/dependencies.ts
Comment (bluwy) [inline packages/create-astro/src/actions/dependencies.ts:53]: The 7 char here seems to indent the CLI for me. Maybe we can rename this as "adapt"?

image

```suggestion
				label: title('adapt'),
```

File: examples/basics/src/layouts/Layout.astro
Comment (bluwy) [inline examples/basics/src/layouts/Layout.astro:8]: nit: Maybe we can use a nicer name like 
```suggestion
		<title>Astro Basics</title>
```

which matches the blog template:
https://github.com/withastro/astro/blob/731b69712b0e9c5d5a4058db3fa3e8bf628c7aad/examples/blog/src/consts.ts#L4

https://github.com/withastro/astro/blob/731b69712b0e9c5d5a4058db3fa3e8bf628c7aad/examples/blog/src/pages/index.astro#L8-L12

**Review by bluwy — APPROVED** [review]
LGTM. Looks like the running the basics template locally causes some dynamic import errors with Vite. But I think that could be investigated separately in the future.

File: packages/create-astro/src/actions/dependencies.ts
Comment (Princesseuh) [reply packages/create-astro/src/actions/dependencies.ts:53]: adpt perhaps? it matches template

Comment (bluwy) [reply packages/create-astro/src/actions/dependencies.ts:53]: I'm fine either ways 👍

**Pull request merged**
